### PR TITLE
issues/1243 - Introduce affwp_dashboard_tabs filter

### PR DESCRIPTION
**Pull request by sumobi**
#1243

### CODE REVIEW

File: includes/misc-functions.php [hunk 1]
```diff
@@ -493,3 +493,61 @@ function affwp_abs_number_round( $val, $precision = 2 ) {
...
+	return apply_filters( 'affwp_affiliate_area_url', get_permalink( affiliate_wp()->settings->get( 'affiliates_page' ) ) );
+}
+
+/**
+ * Add a new tab
+ *
+ * @since  1.7.18
+ * @return string
+ */
+function affwp_tab( $tab = '', $label = '' ) {
```
Comment (pippinsplugins) [inline includes/misc-functions.php]: Let's name this so it's specific to frontend / affliate area.

File: includes/misc-functions.php [hunk 2]
```diff
@@ -493,3 +493,61 @@ function affwp_abs_number_round( $val, $precision = 2 ) {
...
 }
+
+/**
+ * Retrieves the Affiliate Area URL
+ *
+ * @since  1.7.18
+ * @return string URL to the Affiliate Area
+ */
+function affwp_get_affiliate_area_url() {
+	return apply_filters( 'affwp_affiliate_area_url', get_permalink( affiliate_wp()->settings->get( 'affiliates_page' ) ) );
```
Comment (pippinsplugins) [inline includes/misc-functions.php]: I think we added a function for retrieving affiliate area page ID. Can we use that please?

File: includes/misc-functions.php [hunk 3]
```diff
@@ -493,3 +493,61 @@ function affwp_abs_number_round( $val, $precision = 2 ) {
...
+}
+
+/**
+ * Outputs the active CSS class for each tab in the Affiliate Area
+ *
+ * @since  1.7.18
+ * @return string active CSS class
+ */
+function affwp_tabs_active_class( $active_tab, $tab = '' ) {
+	$class = apply_filters( 'affwp_tabs_active_class', 'active' );
```
Comment (pippinsplugins) [inline includes/misc-functions.php]: The tab needs to be passed to the filter.

File: includes/misc-functions.php [hunk 4]
```diff
@@ -493,3 +493,61 @@ function affwp_abs_number_round( $val, $precision = 2 ) {
...
+ */
+function affwp_tab( $tab = '', $label = '' ) {
+
+	$active_tab = ! empty( $_GET['tab'] ) ? sanitize_text_field( $_GET['tab'] ) : 'urls';
+
+	if ( affwp_tabs_show_tab( $tab ) ) : ?>
+	<li class="affwp-affiliate-dashboard-tab<?php echo affwp_tabs_active_class( $active_tab, $tab ); ?>">
+		<a href="<?php echo affwp_tabs_get_url( $tab ); ?>"><?php echo $label; ?></a>
+	</li>
+	<?php endif;
```
Comment (pippinsplugins) [inline includes/misc-functions.php]: I'm not sure I love the tab HTML being in a function. That makes it significantly harder for someone to customize the HTML, even for really simply tasks like adding icons.

File: templates/dashboard.php [hunk 5]
```diff
@@ -29,35 +29,33 @@
...
+
+			<?php echo affwp_tab( 'creatives', __( 'Creatives', 'affiliate-wp' ) ); ?>
+
+			<?php echo affwp_tab( 'settings', __( 'Settings', 'affiliate-wp' ) ); ?>
+
 			<?php do_action( 'affwp_affiliate_dashboard_tabs', affwp_get_affiliate_id(), $active_tab ); ?>
 		</ul>
 
-		<?php affiliate_wp()->templates->get_template_part( 'dashboard-tab', $active_tab ); ?>
+		<?php
```
Comment (pippinsplugins) [inline templates/dashboard.php]: This needs to use the helper function.

File: includes/misc-functions.php (hunk 2 above)
Comment (sumobi) [inline includes/misc-functions.php]: We do, but it's a function specific to the login class. I've changed `get_login_url` to use the new function here: https://github.com/AffiliateWP/AffiliateWP/blob/issue/1243/includes/class-login.php#L167

This way you can filter the individual login URL separately. There are times for example where you might want to login to a different URL.

File: includes/misc-functions.php (hunk 1 above)
Comment (sumobi) [inline includes/misc-functions.php]: Ok

File: includes/misc-functions.php (hunk 3 above)
Comment (sumobi) [inline includes/misc-functions.php]: Good catch

File: includes/misc-functions.php (hunk 4 above)
Comment (sumobi) [inline includes/misc-functions.php]: Hmm good point... Should we kill off the new affwp_tab() function and move this back to the template file?

Comment (sumobi) [inline includes/misc-functions.php]: Or add a filter to this function so you can filter the HTML based on the tab? This way you can still add icons etc, and won't need to edit the entire dashboard.php function to do so. We can provide an example of how to do so.

File: includes/misc-functions.php (hunk 1 above)
Comment (sumobi) [inline includes/misc-functions.php]: done

File: includes/misc-functions.php (hunk 3 above)
Comment (sumobi) [inline includes/misc-functions.php]: done

File: templates/dashboard.php (hunk 5 above)
Comment (sumobi) [inline templates/dashboard.php]: done

File: includes/misc-functions.php (hunk 4 above)
Comment (pippinsplugins) [inline includes/misc-functions.php]: For ease of customization, I think it's important as much of the HTML as possible be in the template files, even if it means a bit of duplication. That is the point of template files after all.

Comment (sumobi) [inline includes/misc-functions.php]: 👍  removed

**Comment by michaelbeil** [comment]
This was added in https://github.com/AffiliateWP/AffiliateWP/pull/1249.

**Pull request closed without merging**